pkg/controller/node/nodecontroller.go

```diff
@@ -238,6 +289,48 @@ func (nc *NodeController) getCondition(status *api.NodeStatus, conditionType api
 	return nil
 }
 
+func (nc *NodeController) checkPod(obj interface{}) {
```

Comment:
this function needs a name that (1) implies it sometimes deletes pods, (2) if possible tries to describe the conditions under which it deletes pods. If (2) is not possible to express concisely you could just call the function maybeDeletePod()

also, please add a function-level comment explaining the reason for each of the deletion circumstances, or put inline comments for each of the blocks where you call nc.forcefullyDeletePod()